shared/src/business/useCaseHelper/caseAssociation/addExistingUserToCase.test.js

```diff
@@ -0,0 +1,125 @@
+const {
+  applicationContext,
+} = require('../../test/createTestApplicationContext');
+const {
+  ROLES,
+  SERVICE_INDICATOR_TYPES,
+} = require('../../entities/EntityConstants');
+const { addExistingUserToCase } = require('./addExistingUserToCase');
+const { Case } = require('../../entities/cases/Case');
+const { MOCK_CASE } = require('../../../test/mockCase');
+
+describe('addExistingUserToCase', () => {
+  const USER_ID = '674fdded-1d17-4081-b9fa-950abc677cee';
+
+  beforeEach(() => {
+    const mockUser = {
+      userId: USER_ID,
+    };
+
+    applicationContext
+      .getPersistenceGateway()
+      .getCognitoUserIdByEmail.mockReturnValue(mockUser);
+
+    applicationContext
+      .getPersistenceGateway()
+      .getUserById.mockReturnValue(mockUser);
+  });
+
+  it('throws an unauthorized error on non admissionsclerk users', async () => {
+    applicationContext.getCurrentUser.mockReturnValue({});
+
+    await expect(
+      addExistingUserToCase({
+        applicationContext,
+        caseEntity: new Case(MOCK_CASE, { applicationContext }),
+        email: 'testing@example.com',
+        name: 'Bob Ross',
+      }),
+    ).rejects.toThrow('Unauthorized');
+  });
+
+  it('throws an error if no user exists with that email', async () => {
+    applicationContext.getCurrentUser.mockReturnValue({
+      role: ROLES.admissionsClerk,
+    });
```

Comment:
This should probably be beforeEach?

Reply:
Some, but not all.

Reply:
One of the tests checks for an empty currentUser.